frontend/packages/dev-console/src/components/import/devfile/devfileHooks.ts

```diff
@@ -107,3 +112,63 @@ export const useDevfileDirectoryWatcher = (
     setFieldValue('docker.dockerfilePath', `${smartSlashDir}Dockerfile`);
   }, [dir, setFieldValue]);
 };
+
+export const useDevfileSource = () => {
+  const searchParams = new URLSearchParams(window.location.search);
+  const devfileSourceUrl = searchParams.get('gitRepo');
+  const devfileName = searchParams.get('devfileName');
+  const formType = searchParams.get('formType');
+  const { setFieldValue } = useFormikContext<FormikValues>();
+
+  React.useEffect(() => {
+    if (devfileSourceUrl) {
+      const gitType = detectGitType(devfileSourceUrl);
+      const gitService = getGitService({ url: devfileSourceUrl }, gitType as GitProvider);
+      setFieldValue('devfile.devfileSourceUrl', devfileSourceUrl);
+      gitService
+        .getDevfileContent()
+        .then((contents) => setFieldValue('devfile.devfileContent', contents))
+        .catch(() => {
+          setFieldValue('devfile.devfileContent', null);
+          setFieldValue('devfile.devfileHasError', true);
+        });
+
+      if (formType === 'sample') {
+        setFieldValue('name', createComponentName(devfileName));
+        setFieldValue('git.url', devfileSourceUrl);
+        setFieldValue('git.type', gitType);
+        setFieldValue('application.initial', SAMPLE_APPLICATION_GROUP);
+        setFieldValue('application.name', SAMPLE_APPLICATION_GROUP);
+        setFieldValue('application.selectedKey', SAMPLE_APPLICATION_GROUP);
+      }
```

Comment:
shouldn't we be setting name, git url and type otherwise as well here as even for catalog flow we need to update it?

```suggestion
        setFieldValue('name', createComponentName(devfileName));
        setFieldValue('git.url', devfileSourceUrl);
        setFieldValue('git.type', gitType);
      if (formType === 'sample') {
        setFieldValue('application.initial', SAMPLE_APPLICATION_GROUP);
        setFieldValue('application.name', SAMPLE_APPLICATION_GROUP);
        setFieldValue('application.selectedKey', SAMPLE_APPLICATION_GROUP);
      }
```

Reply:
Updated.